cart/views.py

```diff
@@ -6,18 +6,154 @@
 
 from cart.models    import Cart, Order, OrderStatus, AddressInformation
 from user.models    import User
+from user.utils     import login_decorator
 from product.models import Product
 from voucher.models import Voucher
 
 
-class UserCartView(View):
+ORDER_STATUS_CHECK = 'checking_out'
+
+
+class CartProductView(View):
     @login_decorator
     def post(self, request):
-        data       = json.loads(request.body)
-        user_id    = request.user.id
-
-        Cart.objects.create(
-                product_id = 
-                user_id    = user_id,
-                voucher_id = voucher_id
-                )
+        try:
+            data             = json.loads(request.body)
+            product_id       = data.get('product_id')
+            user_id          = request.user.id
+
+            if Order.objects.filter(user_id=user_id, order_status__name=ORDER_STATUS_CHECK, carts__product_id=product_id).exists():
+
+                return JsonResponse({'message': False}, status=403)
+
+            if Order.objects.filter(user_id=user_id, order_status__name=ORDER_STATUS_CHECK).exists() and not Cart.objects.filter(product_id=product_id, order__order_status__name=ORDER_STATUS_CHECK, order__user_id=user_id).exists():
+                Cart.objects.create(
+                    product_id = product_id, 
+                    order_id   = Order.objects.get(user_id=user_id, order_status__name=ORDER_STATUS_CHECK).id
+                    )
+                return JsonResponse({'message': 'SUCCESS'}, status=201)
+
+            if not Order.objects.filter(user_id=user_id, order_status__name=ORDER_STATUS_CHECK).exists():
+                Order.objects.create(
+                        user_id         = user_id,
+                        order_status_id = OrderStatus.objects.get(name=ORDER_STATUS_CHECK).id
+                        )
+
+                Cart.objects.create(
+                        product_id = product_id,
+                        order_id   = Order.objects.get(user_id=user_id, order_status__name=ORDER_STATUS_CHECK).id
+                        )
```

Comment:
Order.objects.create을 order라는 변수에 담으면 생성과 동시에 해당 객체를 바로 사용 할 수 있습니다.
```suggestion
                order = Order.objects.create(
                        user_id         = user_id,
                        order_status_id = OrderStatus.objects.get(name=ORDER_STATUS_CHECK).id
                        )

                Cart.objects.create(
                        product_id = product_id,
                        order_id   = order.id
                        )
```